File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/database/OctopointsDb.kt
package dev.tpavani.octopoints.db.database

import androidx.room.Database
import androidx.room.RoomDatabase
import dev.tpavani.octopoints.db.dao.*
import dev.tpavani.octopoints.db.entity.MatchEntity
import dev.tpavani.octopoints.db.entity.TeamEntity
import dev.tpavani.octopoints.db.entity.TeamUserRelationEntity
import dev.tpavani.octopoints.db.entity.UserEntity

@Database(entities = [UserEntity::class, MatchEntity::class, TeamEntity::class, TeamUserRelationEntity::class], version = 1)
abstract class OctopointsDb : RoomDatabase() {
    abstract fun userDao(): UserDao
    abstract fun matchDao(): MatchDao
    abstract fun teamDao(): TeamDao
    abstract fun teamUserRelationDao(): TeamUserRelationDao
    abstract fun ruleDao(): RuleDao
}

File: octopoints_flutter/lib/db/database/OctopointsDb.dart
import 'package:floor/floor.dart';
import 'package:octopoints_flutter/db/dao/RuleDao.dart';
import 'package:octopoints_flutter/db/dao/TeamUserRelationDao.dart';
import 'package:octopoints_flutter/db/dao/MatchDao.dart';
import 'package:octopoints_flutter/db/dao/TeamDao.dart';
import 'package:octopoints_flutter/db/dao/UserDao.dart';
import 'package:octopoints_flutter/db/entity/MatchEntity.dart';
import 'package:octopoints_flutter/db/entity/RuleEntity.dart';
import 'package:octopoints_flutter/db/entity/TeamEntity.dart';
import 'package:octopoints_flutter/db/entity/TeamUserRelation.dart';
import 'package:octopoints_flutter/db/entity/UserEntity.dart';
import 'dart:async';
import 'package:sqflite/sqflite.dart' as sqflite;

part 'OctopointsDb.g.dart';

@Database(
  version: 1,
  entities: [MatchEntity, UserEntity, TeamEntity, TeamUserRelation, RuleEntity],
)
abstract class OctopointsDb extends FloorDatabase {
  MatchDao get matchDao;
  UserDao get userDao;
  TeamDao get teamDao;
  TeamUserRelationDao get teamUserRelationDao;
  RuleDao get ruleDao;
}
